### PR TITLE
Added RemoveReporCaches class

### PR DESCRIPTION
**Pull request by Jestemrique**
Added new sample on how to remove a report's cache.
Created package to add other classes for other operations / object types in the future

### CODE REVIEW

File: stand-alone/com/microstrategy/samples/cachemanager/RemoveReportCaches.java [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+		// Retrieve reports caches for project.
+		Caches reportCaches = getReportCaches(iServerSession);
+		System.out.println("Retrieving report caches for project: " + reportCaches.getProjectName());
+		System.out.println("Caches found: " + reportCaches.getCount());
+
+		// Deleting caches for specific report.
+		System.out.println("Deleting cache for report: " + reportID);
+		deleteReportCaches(iServerSession, reportID);
+		System.out.println("Cache deleted.");
+
```
Comment (tiagosiebler) [inline stand-alone/com/microstrategy/samples/cachemanager/RemoveReportCaches.java:48]: trim line

File: stand-alone/com/microstrategy/samples/cachemanager/RemoveReportCaches.java [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+	}
+
+	// Delete caches for report identified by reportID
+	public static void deleteReportCaches(WebIServerSession iServerSession, String reportID) {
+		int monitorType = EnumWebMonitorType.WebMonitorTypeCache;
+		int cacheInfo = EnumDSSXMLCacheInfo.DssXmlCacheInfoReportId;
+
+		removeCachesFromObject(iServerSession, reportID, monitorType, cacheInfo);
+
```
Comment (tiagosiebler) [inline stand-alone/com/microstrategy/samples/cachemanager/RemoveReportCaches.java:57]: trim line

File: stand-alone/com/microstrategy/samples/cachemanager/RemoveReportCaches.java [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+		CacheSource cacheSource = (CacheSource) factory.getMonitorSource(monitorType);
+		cacheSource.setLevel(EnumDSSXMLLevelFlags.DssXmlDetailLevel);
+
+		try {
+			CacheResults cacheResults = cacheSource.getCaches();
+			projectCaches = cacheResults.item(iServerSession.getProjectID());
+		} catch (WebObjectsAdminException | WebObjectsException | IllegalArgumentException e) {
+			e.printStackTrace();
+		}
+
```
Comment (tiagosiebler) [inline stand-alone/com/microstrategy/samples/cachemanager/RemoveReportCaches.java:81]: trim line

File: stand-alone/com/microstrategy/samples/cachemanager/RemoveReportCaches.java [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+		return projectCaches;
+
```
Comment (tiagosiebler) [inline stand-alone/com/microstrategy/samples/cachemanager/RemoveReportCaches.java:83]: trim line

File: stand-alone/com/microstrategy/samples/cachemanager/RemoveReportCaches.java [hunk 5]
```diff
@@ -0,0 +1,105 @@
...
+		CacheManipulator cacheManipulator = cacheSource.getManipulator();
+
+		try {
+			MonitorFilter monitorFilter = cacheManipulator.newMonitorFilter();
+			monitorFilter.add(cacheInfo, EnumDSSXMLMonitorFilterOperator.DssXmlEqual, objectID);
+			cacheManipulator.alter(iServerSession.getProjectID(), EnumDSSXMLCacheAdminAction.DssXmlDeleteCache, monitorFilter);
+		} catch (WebObjectsAdminException | MonitorManipulationException | WebObjectsException | IllegalArgumentException e) {
+			e.printStackTrace();
+		}
+
```
Comment (tiagosiebler) [inline stand-alone/com/microstrategy/samples/cachemanager/RemoveReportCaches.java:103]: trim line

File: stand-alone/com/microstrategy/samples/cachemanager/RemoveReportCaches.java (hunk 1 above)
Comment (Jestemrique) [reply stand-alone/com/microstrategy/samples/cachemanager/RemoveReportCaches.java:48]: Done!

File: stand-alone/com/microstrategy/samples/cachemanager/RemoveReportCaches.java (hunk 2 above)
Comment (Jestemrique) [reply stand-alone/com/microstrategy/samples/cachemanager/RemoveReportCaches.java:57]: Done!

File: stand-alone/com/microstrategy/samples/cachemanager/RemoveReportCaches.java (hunk 3 above)
Comment (Jestemrique) [reply stand-alone/com/microstrategy/samples/cachemanager/RemoveReportCaches.java:81]: Done!

File: stand-alone/com/microstrategy/samples/cachemanager/RemoveReportCaches.java (hunk 4 above)
Comment (Jestemrique) [reply stand-alone/com/microstrategy/samples/cachemanager/RemoveReportCaches.java:83]: Done!

File: stand-alone/com/microstrategy/samples/cachemanager/RemoveReportCaches.java (hunk 5 above)
Comment (Jestemrique) [reply stand-alone/com/microstrategy/samples/cachemanager/RemoveReportCaches.java:103]: Done!

**Pull request merged**